core/trino-main/src/main/java/io/trino/cost/ComparisonStatsCalculator.java

```diff
@@ -239,4 +249,61 @@ private static PlanNodeStatsEstimate estimateExpressionNotEqualToExpression(
         rightExpressionSymbol.ifPresent(symbol -> result.addSymbolStatistics(symbol, rightNullsFiltered));
         return result.build();
     }
+
+    private static PlanNodeStatsEstimate estimateExpressionToExpressionInequality(
+            ComparisonExpression.Operator operator,
+            PlanNodeStatsEstimate inputStatistics,
+            SymbolStatsEstimate leftExpressionStatistics,
+            Optional<Symbol> leftExpressionSymbol,
+            SymbolStatsEstimate rightExpressionStatistics,
+            Optional<Symbol> rightExpressionSymbol,
+            boolean isUnestimatedFilterCostApproximationEnabled)
+    {
+        if (leftExpressionStatistics.isUnknown() || rightExpressionStatistics.isUnknown()) {
+            return PlanNodeStatsEstimate.unknown();
+        }
+
+        StatisticRange leftExpressionRange = StatisticRange.from(leftExpressionStatistics);
+        StatisticRange rightExpressionRange = StatisticRange.from(rightExpressionStatistics);
+        if (leftExpressionRange.isEmpty() || rightExpressionRange.isEmpty()) {
+            return PlanNodeStatsEstimate.unknown();
+        }
+
+        switch (operator) {
+            case LESS_THAN:
+            case LESS_THAN_OR_EQUAL:
+                // left is always lesser than right
+                if (leftExpressionRange.getHigh() < rightExpressionRange.getLow()) {
+                    PlanNodeStatsEstimate.Builder estimate = PlanNodeStatsEstimate.buildFrom(inputStatistics);
+                    leftExpressionSymbol.ifPresent(symbol -> estimate.addSymbolStatistics(symbol, leftExpressionStatistics));
+                    rightExpressionSymbol.ifPresent(symbol -> estimate.addSymbolStatistics(symbol, rightExpressionStatistics));
+                    return estimate.build();
```

Comment:
consider extracting lines 277-280 to a method and reuse it in `GREATER_THAN_OR_EQUAL` case.

Reply:
extracting to a method results in a long list of arguments to the methods, so keeping it as is for now